include/deal.II/numerics/vector_tools.h

```diff
@@ -924,6 +928,26 @@ namespace VectorTools
                 VectorType &vec_result);
 
   /**
+   * Generic implementation for the project() function on parallel Triangulations with
+   * FiniteElements that are supported by the MatrixFree class.
+   * Use this function if you use more than four components or the degree of your FiniteElement
+   * is higher than eight. You have to specify the first template arguments yourself.
+   * @p vec_result is expected to not have any ghost entries.
+   */
+  template <int components, int fe_degree, int dim, typename VectorType, int spacedim>
+  void project_generic (const Mapping<dim, spacedim>                              &mapping,
```

Comment:
I'm not sure how much we should expose. If we want a user to be able to use any of the 'unsupported' combinations, we need to expose all parallel functions. My plan was to have a fall-back that does not use templates (at least for the polynomial degree and the number of quadrature points), which comes in with the larger matrix-free update.